common/djangoapps/pipeline_mako/templates/static_content.html

```diff
@@ -86,6 +86,47 @@
 source, template_path = Loader(engine).load_template_source(path)
 %>${source | n, decode.utf8}</%def>
 
+<%def name="studiofrontend(page, lang='en')">
+    <%doc>
+        Loads a studio-frontend page, with the necessary context. Context is expected
+        as a dictionary in the body of this tag.
+
+        Dev note: we could also add the locale-injection script in this block
+            -use a better default than the hardcoded 'en'. There should be a setting or something?
+            -lookup (webpack exported) locale-injection script using lang as key
+            -include it as the first script in this block
+    </%doc>
+    <%
+        from django.template import Template, Context
+        from webpack_loader.exceptions import WebpackLoaderBadStatsError
+        import json
+
+        def _erics_hacky_helper(input_dict):
+            # dumps a dict to json in a way that can be consumed
+            ret = "{"
+            print input_dict
+            for key in input_dict:
+                ret = "{} {}: \"{}\",".format(ret, key, input_dict[key])
+            ret += "}"
+            return ret
+
+        body = capture(caller.body)
+        body_dict = json.loads(body)
+        body_dict['lang'] = lang
+        return Template("""
+            <script type="text/javascript" id='courseContext'>
+                var courseContext = {% autoescape off %}{{ body }}{% endautoescape %};
+            </script>
+            <div id="root"></div>
```

Comment:
Key insight from @fysheets - the context var and root div both need to be defined in the DOM before running any of our sfe scripts.